src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,213 @@
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id AS car_id, c.model, m.id AS manufacturer_id, m.name, "
+                + "m.country FROM cars c "
+                + "LEFT JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> listCars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllCarStatement
+                         = connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllCarStatement.executeQuery();
+            while (resultSet.next()) {
+                listCars.add(getCarWithoutDrivers(resultSet));
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Couldn't get all cars", ex);
+        }
+        for (Car car : listCars) {
+            car.setDrivers(getAllDriversByCar(car));
+        }
+        return listCars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                   PreparedStatement updateCarStatement
+                           = connection.prepareStatement(query)) {
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setLong(2, car.getManufacturer().getId());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in carsDB.", ex);
+        }
+        if (car.getDrivers() != null) {
+            addAllDriversByCar(car);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                   PreparedStatement updateCarStatement
+                            = connection.prepareStatement(query)) {
+            updateCarStatement.setLong(1, id);
+            int i = updateCarStatement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Couldn't delete car with id: "
+                    + id, ex);
+        }
+        hardDeleteAllDriversByCar(id);
+        return true;
+    }
+
+    @Override
+    public List<Car> getAllCarsByDriver(Long driverId) {
+        String query = "SELECT c.id AS car_id, c.model, c.manufacturer_id as manufacturer_id, "
+                + "m.name, m.country, cd.driver_id AS driver_id \n"
+                + "FROM cars c \n"
+                + "LEFT JOIN manufacturers m ON manufacturer_id = m.id\n"
+                + "LEFT JOIN cars_drivers cd ON c.id = cd.car_id\n"
```

Comment:
Same here

Reply:
I need all data from the first table, so that's why LEFT JOIN